### PR TITLE
[Bugfix][V1] Chunk prompt-logprobs logits to bound the activation peak

### PR DESCRIPTION
**Pull request by malaiwah**
## Purpose

`_get_prompt_logprobs_dict` materializes the full `[num_scheduled_tokens, vocab_size]` logits tensor in one shot and then upcasts it to float32 via `log_softmax`. Both tensors are live simultaneously, so the transient peak is roughly:

```
num_scheduled_tokens * vocab_size * (2 bytes bf16 + 4 bytes fp32)
```

At 8192 scheduled tokens and a 152k vocabulary that is **~7.5 GiB**.

That peak is never observed by memory profiling. `profile_run` calls `_dummy_sampler_run`, which constructs its `SamplingMetadata` with:

https://github.com/vllm-project/vllm/blob/main/vllm/v1/worker/gpu_model_runner.py#L6362

```python
max_num_logprobs=None,
```

so the logprobs path is not exercised and the float32 score tensor is never allocated during profiling. KV cache capacity is sized against a peak that excludes prompt logprobs, and the allocation has to come out of whatever is left at serving time.

This is exactly the mechanism @robertgshaw2-redhat described in #5907:

> When determining the KV cache size, we calculate peak memory running a long prefill *without logprobs*. If a prompt requests many logprobs, however, this is an additional source of memory usage which is not considered during warmup and can cause OOM because we have nothing in scheduler to prevent this.

## Approach

Rather than restructure profiling, this bounds the allocation directly — the "chunked logits processing" approach proposed by @tjohnson31415 in #5907 and endorsed in that thread as "the ideal solution".

`compute_logits` and the score computation now run over row chunks of `VLLM_PROMPT_LOGPROBS_CHUNK_SIZE` (default 1024). Each chunk's results are copied into the existing per-request CPU tensors before the next chunk is materialized, so the peak becomes independent of `max_num_batched_tokens`.

Behaviour is unchanged: same logits, same scores, same destination rows, same `logprobs_mode` handling. Only the number of rows materialized at once differs. The default of 1024 leaves the common case (short prompts, small `max_num_batched_tokens`) at a single chunk, so there is no change in behaviour or performance for most deployments.

## Why this is not duplicating an existing PR

Checks run:

```
gh issue view 5907 --repo vllm-project/vllm --comments
gh pr list --repo vllm-project/vllm --state open --search "prompt_logprobs memory"
gh pr list --repo vllm-project/vllm --state open --search "prompt logprobs profile_run"
gh pr list --repo vllm-project/vllm --state open --search "prompt_logprobs OOM"
gh pr list --repo vllm-project/vllm --state merged --search "5907 in:body"
```

- **#5907** — the tracking issue for this exact defect. Closed as **stale**, never fixed.
- **#5355** ("[Bugfix] Take the VRAM usage of prompt_logprobs into account") — the referenced fix attempt. **Closed, unmerged.**
- No merged PR addresses it. Verified the behaviour is still present on `main` at `23f360edaa`.
- The four open `prompt_logprobs` PRs are unrelated concerns: **#51846** (`prompt_logprobs=0` admission guards / chat echo), **#49622** (`kv_sharing_fast_prefill` incompatibility validation), **#42245** (non-determinism with prefix caching), **#51953** (DiffusionGemma tiled logits projection — a different model's sampler path).

## Test commands run and results

New test file `tests/v1/worker/test_gpu_prompt_logprobs_chunking.py`, modelled on the existing `tests/v1/worker/test_gpu_rejection_sampler_chunking.py` convention:

```
python -m pytest tests/v1/worker/test_gpu_prompt_logprobs_chunking.py -v
9 passed
```

Coverage — chunked output must match a single-chunk reference exactly, across `chunk_size` ∈ {1, 3, 4, 1024} (degenerate, short-final-chunk, evenly-dividing, single-chunk) and `logprobs_mode` ∈ {`raw_logprobs`, `raw_logits`}, plus an assertion that no single `compute_logits` call exceeds the configured chunk size and that every row is covered exactly once.

**Verified by negation:** injecting the classic chunking bug (destination slice ignoring the chunk offset, `slice(start_idx, start_idx + (end - pos))`) makes 6 of the 9 tests fail with a 100%-mismatched tensor. The 3 that still pass are the single-chunk cases, where the bug is unreachable. So the tests genuinely pin the row mapping rather than passing vacuously.

The tests are CPU-runnable. They disable `PIN_MEMORY` via an autouse fixture because `async_tensor_h2d` pins the source buffer, which requires a CUDA allocator; the chunking logic under test is unaffected.

## AI assistance

This change was developed with AI assistance. The problem analysis, the upstream-vs-fork verification, the patch and the tests were produced with an AI coding agent; the duplicate-work checks above were run as part of that process. A human submitter *has* reviewed every changed line.

The chunking approach is a port of a fix that has been running in production in a downstream fork, where it resolved this OOM.

### CODE REVIEW

**Review by malaiwah — COMMENTED** [review]
`VLLM_PROMPT_LOGPROBS_CHUNK_SIZE` is a safe defense to prevent OOM (DOS on the vLLM endpoint)

**Review by njhill — CHANGES_REQUESTED** [review]
Thanks @malaiwah ... but please use V2 model runner which already addresses this.

We are trying to minimize work on V1 model runner now, V2 is ready for general use.

`VLLM_USE_V2_MODEL_RUNNER=1`
